pkg/applyconfigurations/applyconfiguration_integration_test.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package applyconfigurations
+
+import (
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"sigs.k8s.io/controller-tools/pkg/crd"
+	"sigs.k8s.io/controller-tools/pkg/genall"
+	"sigs.k8s.io/controller-tools/pkg/loader"
+	"sigs.k8s.io/controller-tools/pkg/markers"
+)
+
+type outputToMap map[string]*outputFile
+
+// Open implements genall.OutputRule.
+func (m outputToMap) Open(_ *loader.Package, path string) (io.WriteCloser, error) {
+	if _, ok := m[path]; !ok {
+		m[path] = &outputFile{}
+	}
+	return m[path], nil
+}
+
+type outputFile struct {
+	contents []byte
+}
+
+func (o *outputFile) Write(p []byte) (int, error) {
+	o.contents = append(o.contents, p...)
+	return len(p), nil
+}
+
+func (o *outputFile) Close() error {
+	return nil
+}
+
+var _ = Describe("CRD Generation From Parsing to CustomResourceDefinition", func() {
+	outputPath := "testapplyconfiguration"
+	masterPath := "applyconfiguration"
+	By("resetting the test directory")
+	// not using after each to allow for inspection of failed tests
+	BeforeEach(func() {
+		Expect(os.RemoveAll(filepath.Join(".", outputPath))).To(Succeed())
+	})
+	It("should be able to verify generated ApplyConfiguration types for the CronJob schema", func() {
+		By("switching into testdata to appease go modules")
+		cwd, err := os.Getwd()
+		Expect(err).NotTo(HaveOccurred())
+		Expect(os.Chdir("./testdata/cronjob")).To(Succeed()) // go modules are directory-sensitive
+		defer func() { Expect(os.Chdir(cwd)).To(Succeed()) }()
+
+		output := make(outputToMap)
+
+		By("initializing the runtime")
+		optionsRegistry := &markers.Registry{}
+		Expect(optionsRegistry.Register(markers.Must(markers.MakeDefinition("crd", markers.DescribesPackage, crd.Generator{})))).To(Succeed())
+		// Add the applyconfigurations generator but set it to verify only.
+		// This allows us to check if there's a diff between the checked in generated data and what it would generate now.
```

Comment:
Is this correct? We seem to make the output var and set it on output rules, but I don't think that's actually being used later 🤔

Reply:
I'm not following :)